framework/wazuh/rbac/orm.py

```diff
@@ -2171,25 +2175,29 @@ def add_rule_to_role(self, rule_id: int, role_id: int, atomic: bool = True):
             This parameter indicates if the operation is atomic. If this function is called within
             a loop or a function composed of several operations, atomicity cannot be guaranteed.
             And it must be the most external function that ensures it
+        force_admin : bool
+            By default, changing an administrator roles is not allowed. If True, it will be applied to admin roles too
 
         Returns
         -------
         True -> Success | False -> Failure | Role not found | Rule not found | Existing relationship
         """
         try:
             # Create a rule-role relationship if both exist
-            rule = self.session.query(Rules).filter_by(id=rule_id).first()
-            if rule is None:
-                return SecurityError.RULE_NOT_EXIST
-            role = self.session.query(Roles).filter_by(id=role_id).first()
-            if role is None:
-                return SecurityError.ROLE_NOT_EXIST
-            if self.session.query(RolesRules).filter_by(rule_id=rule_id, role_id=role_id).first() is None:
-                role.rules.append(rule)
-                atomic and self.session.commit()
-                return True
-            else:
-                return SecurityError.ALREADY_EXIST
+            if int(rule_id) > max_id_reserved or force_admin:
```

Comment:
We should raise a different exception when this condition is not met (default rules)